core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedStageExecution.java

```diff
@@ -395,7 +396,11 @@ private synchronized boolean isStageFlushing()
 
     private synchronized boolean isStageFinished()
```

Comment:
Can we do similar optimization in isStageFlushing ?

Reply:
`isStageFlushing` we check two different sets. There is a high chance that flushing size and finished size will be higher than `allTask` size, and this will not mean that all tasks are flushing or finishing.

Reply:
Right, if we can have a task id in both sets at some point in time, then we can't make similar optimization there.